File: content/tracks/dsa/problems/lc-0128-longest-consecutive-sequence/Solution.java
import java.util.*;

class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> values = new HashSet<>();
        for (int num : nums) {
            values.add(num);
        }
        int longest = 0;
        for (int num : values) {
            if (values.contains(num - 1)) {
                continue; // not the start of a run: its run is counted from the smallest value
            }
            int length = 1;
            while (values.contains(num + length)) {
                length++;
            }
            longest = Math.max(longest, length);
        }
        return longest;
    }
}

File: content/tracks/dsa/problems/lc-0128-longest-consecutive-sequence/solution.py
from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        values = set(nums)
        longest = 0
        for num in values:
            if num - 1 in values:
                continue  # not the start of a run: its run is counted from the smallest value
            length = 1
            while num + length in values:
                length += 1
            longest = max(longest, length)
        return longest
